### PR TITLE
Fix incorrect byte comparison

### LINKED ISSUE #31700: Bad code generated for byte comparison
This prints `false` when it should print `true`.

```
import ballerina/io;

public function main() {
    io:println(testByte(0x20));
}

function testByte(byte b) returns boolean {
    if b >= 0x20 {
       return true;
    }
    return false;
}
```

Changing `byte` to `int` or changing `0x20` to decimal both make the problem go away.

### PR DESCRIPTION
**Pull request by Nadeeshan96**
Fixes #31700

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/greater-less-than-operation.bal [hunk 1]
```diff
@@ -24,6 +24,27 @@ function testFloatRanges(float a) returns (int) {
     return retunType;
 }
 
+function testByteComparison() {
+    byte lowValue = 0x20;
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/greater-less-than-operation.bal:28]: Shall we same test with byte value with decimals.

Comment (Nadeeshan96) [reply tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/greater-less-than-operation.bal:28]: I ran the same test with byte value with decimals. It was fine. Do I need to add those lines here also.

Comment (warunalakshitha) [reply tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/greater-less-than-operation.bal:28]: I mean lets keep the hexa decimal value as well.

something like

function testByteComparison() {
compareByteValues(0x20, 0x21);
compareByteValues(32, 45);
compare ByteValues(0x20, 35);
}

compare ByteValues(lowValue, highValue ) {// your compare test}
}

Comment (Nadeeshan96) [reply tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/greater-less-than-operation.bal:28]: I will look into this.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**